File: Sources/Domain/TrafficLedgerObservation.swift
import Foundation

struct TrafficLedgerObservation: Equatable, Sendable {
  let observedAt: Date
  let kernelTotal: TrafficBytes
  let transition: TrafficLedgerTransition
}

enum TrafficLedgerTransition: Equatable, Sendable {
  case baselineEstablished
  case delta(TrafficDeltaReport)
  case countersReset
}

File: platform/windows/MihomoMeter.Windows.Core/Domain/TrafficLedgerObservation.cs
namespace MihomoMeter.Windows.Core.Domain;

public sealed record TrafficLedgerObservation(
    DateTimeOffset ObservedAt,
    TrafficBytes KernelTotal,
    TrafficLedgerTransition Transition);

public abstract record TrafficLedgerTransition;

public sealed record TrafficLedgerBaselineEstablished : TrafficLedgerTransition;

public sealed record TrafficLedgerDelta(TrafficDeltaReport Report) : TrafficLedgerTransition;

public sealed record TrafficLedgerCountersReset : TrafficLedgerTransition;
